airbyte-integrations/connectors/source-presta-shop/source_presta_shop/streams.py

```diff
@@ -0,0 +1,679 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+from abc import ABC
+from datetime import datetime
+from typing import Any, Iterable, Mapping, MutableMapping, Optional
+
+import requests
+from airbyte_cdk.sources.streams.http import HttpStream
+
+
+class PrestaShopStream(HttpStream, ABC):
+    """
+    PrestaShop API Reference: https://devdocs.prestashop.com/1.7/basics/introduction/
+    """
+
+    primary_key = "id"
+    data_key = None
+    limit = 50
+
+    def __init__(self, url: str, **kwargs):
+        super(PrestaShopStream, self).__init__(**kwargs)
+        self._url = url
+        self._offset = 0
+        self._has_next_page = False
+
+    @property
+    def url_base(self) -> str:
+        return f"{self._url}/api/"
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return self._has_next_page
```

Comment:
I don't get it, why we introduce state variable like `_has_next_page` and `_offset`?
why we don't return 
```python
{
  "limit":  f"{self._offset},{self.limit}"
}
```
here?

Reply:
also it is hard to say how response looks like, without running test container